res/controllers/Denon-MC7000-scripts.js

```diff
@@ -816,25 +849,78 @@ MC7000.sortLibrary = function(channel, control, value) {
 // VuMeters only for Channel 1-4 / Master is on Hardware
 MC7000.VuMeter = function(value, group) {
     var deckNumber = script.deckFromGroup(group),
-        vuLevelOutValue = engine.getValue(group, "PeakIndicator") ? MC7000.VuMeterLEDPeakValue : Math.pow(value, 4) * (MC7000.VuMeterLEDPeakValue - 1);
-
-    midi.sendShortMsg(0xB0 + deckNumber - 1, 0x1F, vuLevelOutValue);
+        currentLED = Math.pow(value, 4) * (MC7000.VuMeterLEDPeakValue - 1),
+        vuLevelOutValue = engine.getValue(group, "PeakIndicator") ? MC7000.VuMeterLEDPeakValue : currentLED,
+        vuThreshold = Math.floor(currentLED / 6) * 6;
+    // only send new Midi signal when playing and threshold is met
+    if (engine.getValue(group, "play")) {
```

Comment:
No - the VU meter shall give me a signal while playing - e.g. during cuing in a new track - with Volume at zero. I only introduced that dependency as with the threshold it happens that the lowest LED remains lit and don't switch off when the last signal was below threshold. That is a strange situation to see  the VU still showing one LED while nothing is playing.

Reply:
I get your point though - need to think about it a bit more

Reply:
As the user of the controller I'd expect it to reflect the mixxx VU at all times. I don't really see the usecase for the vumeter outputting even though the channel is silent. That could be quite confusing for people that don't know that this is the standard behavior.

If the lowest led remains lit there is probably rather a bug in your math and the `play` CO is just a workaround for that bug.

Reply:
Yes you are right - I untroduced the condition `play` to overcome the last LED not showing the signal. My calculation is right though. Thinking now a better way is to check for signal < threshold and if so then show nothing, otherwise the signal.

Reply:
Thanks. But I already found another solution. Don't waste your time for me. Cheers!

Reply:
> As the user of the controller I'd expect it to reflect the mixxx VU at all times. I don't really see the usecase for the vumeter outputting even though the channel is silent.

I think you misunderstand the point here - talking about channel vu meter - the master is handled by the controller per hardware and not reflected at all within Mixxx. For the Channels it is standard to show the full signal even there is no sound coming out the master output. So the channel VU shows full level and Master could be silent with VU meter not showing anything. This is the same behavior in Mixxx GUI

Reply:
Ah yes, my mistake. You are right.